src/BowlingScorecard.js

```diff
@@ -0,0 +1,143 @@
+// const Frame = require('./Frame.js');
+
+class BowlingScorecard {
+
+  constructor() {
+    this.frame = 1
+    this.frames = [new Frame(this.frame)]
+  }
+
+  enterRoll(score) {
+    if (this._isInvalidScore(score)) return `Invalid score entered, score must be between 0 and ${this._maxRoll()}.`;
+
+    score = Number(score)
+    this._currentFrame().enterRoll(score)
+    this._applyBonus()
+    if (this._checkFrameOver() === "EndGame") return "End of Game";
+
+    return score
+  }
+
+  currentScore() {
+    let total = 0
+    this.frames.forEach(function(element) {
+      total += (element.frameScore)
+    })
+
+    return total
+  }
+
+  generateScorecardInfo() {
+    let thisIs = this
+    let results = this.frames.map(function(element) {
+      let firstRoll = thisIs._tidyDisplay(element.firstRoll, element.strike)
+      let secondRoll = thisIs._tidyDisplay(element.secondRoll, element.strike, element.spare)
+
+      if (element.frame === 10) {
+        let thirdRoll = thisIs._tidyDisplay(element.thirdRoll, element.strike, element.spare, true)
+        return { frame: element.frame, firstRoll: firstRoll, secondRoll: secondRoll, thirdRoll: thirdRoll, strike: element.strike, spare: element.spare, total: element.total };
+      }
+
+      return { frame: element.frame, firstRoll: firstRoll, secondRoll: secondRoll, strike: element.strike, spare: element.spare, total: element.total };
+    });
+    return results
+  }
+
+  _isInvalidScore(score) {
+    if (score > this._maxRoll() || score < 0) return true
+    if (`${score}`.match(/\D/) !== null) return true
+
+    return false
+  }
+
+  _maxRoll() {
+    if (this._currentFrame().isFirstRollComplete() && this.frame === 10) {
+      if (this._currentFrame().secondRoll === null) {
+        return 10
+      }
+    } else if (this.frame === 10 && this._currentFrame.firstRoll !== null) {
+      if (this._currentFrame().secondRoll === 10) {
+        return 10
+      } else {
+        return 10 - this._currentFrame().secondRoll
+      }
+    }
+
+    let roll = this._currentFrame().firstRoll
+    if (roll !== null) return (10 - roll)
+
+    return 10
+  }
+
+  _checkFrameOver() {
+    this._updateTotal()
+
+    if (this.frame === 10 && this._currentFrame.firstRoll !== null && this._currentFrame.secondRoll !== null && this._currentFrame.thirdRoll !== null ) return "EndGame";
+    if (this.frame === 10 && this._currentFrame.firstRoll !== null && this._currentFrame.secondRoll !== null && this._currentFrame.strike === false && this._currentFrame.spare === false) return "EndGame";
+
```

Comment:
to DRY your code, perhaps you could extract 'this.frame === 10 && this._currentFrame.firstRoll !== null && this._currentFrame.secondRoll !== null' into a method